### PR TITLE
Backend config

### PR DESCRIPTION
**Pull request by ahmedhamilastage**
-isort conf 
-doctsring 
-environ replace decouple
-precommit

### CODE REVIEW

File: app/users/services/__init__.py
Comment (MohamedMegdich1) [inline app/users/services/__init__.py:1]: remove this

File: app/users/serializers/__init__.py
Comment (MohamedMegdich1) [inline app/users/serializers/__init__.py:1]: remove this

File: app/users/models/__init__.py
Comment (MohamedMegdich1) [inline app/users/models/__init__.py:1]: remove this

File: app/users/controllers/__init__.py
Comment (MohamedMegdich1) [inline app/users/controllers/__init__.py:1]: remove this

File: app/users/admin/__init__.py
Comment (MohamedMegdich1) [inline app/users/admin/__init__.py:1]: remove this

File: app/cms/serializers/__init__.py
Comment (MohamedMegdich1) [inline app/cms/serializers/__init__.py:1]: remove this

File: app/cms/models/__init__.py
Comment (MohamedMegdich1) [inline app/cms/models/__init__.py:1]: remove this

File: app/announcement/serializers/__init__.py
Comment (MohamedMegdich1) [inline app/announcement/serializers/__init__.py:1]: remove this

File: app/announcement/models/__init__.py
Comment (MohamedMegdich1) [inline app/announcement/models/__init__.py:1]: remove this

File: app/announcement/controllers/__init__.py
Comment (MohamedMegdich1) [inline app/announcement/controllers/__init__.py:1]: remove this

File: app/announcement/admin/__init__.py
Comment (MohamedMegdich1) [inline app/announcement/admin/__init__.py:1]: remove this

File: app/announcement/admin/utils/ProjectImageAdmin.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
 from django.contrib import admin
+
 from app.announcement.models.utils.ProjectImage import ProjectImage
 
+
 class ProjectImageAdmin(admin.ModelAdmin):
-    list_display = ('id','id', 'image')
+    list_display = ("id", "id", "image")
```
[the remark this answers is not in the document]
Comment (ahmedhamilastage) [reply app/announcement/admin/utils/ProjectImageAdmin.py]: fixed in last commit

File: app/users/controllers/ArchimatchUserViewSet.py [hunk 2]
```diff
@@ -1,38 +1,116 @@
...
+    View for obtaining JSON Web Tokens for ArchimatchUser.
+
+    Inherits from TokenObtainPairView and uses ArchimatchUserObtainPairSerializer
+    for token retrieval.
+    """
 
+    serializer_class = ArchimatchUserObtainPairSerializer
 
 
 class ArchimatchUserViewSet(viewsets.ModelViewSet):
```
Comment (MohamedMegdich1) [inline app/users/controllers/ArchimatchUserViewSet.py:35]: why using ModelViewSet change it

File: app/users/controllers/ClientViewSet.py [hunk 3]
```diff
@@ -1,20 +1,70 @@
...
+    """
+
     serializer_class = ClientSerializer
     queryset = Client.objects.all()
-    
 
-    @action(detail=False, methods=['post'], permission_classes=[], name='login',serializer_class=UserAuthSerializer)
+    @action(
+        detail=False,
+        methods=["post"],
```
Comment (MohamedMegdich1) [inline app/users/controllers/ClientViewSet.py:34]: uppercase

File: app/cms/urls.py [hunk 4]
```diff
@@ -1,11 +1,12 @@
-from django.urls import path, include
+from django.urls import include, path
+
 from rest_framework import routers
+from rest_framework_simplejwt.views import TokenObtainPairView, TokenRefreshView
```
Comment (MohamedMegdich1) [inline app/cms/urls.py:4]: why importing this ?

File: app/cms/urls.py [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
 
-from rest_framework_simplejwt.views import TokenRefreshView, TokenObtainPairView
 from app.cms.controllers import BlogViewSet
 
 router = routers.DefaultRouter()
-router.register('blog',BlogViewSet,basename='blog')
+router.register("blog", BlogViewSet, basename="blog")
```
Comment (MohamedMegdich1) [inline app/cms/urls.py:9]: Please remember the following text: "Represent routes in a tree."

File: app/cms/serializers/__init__.py [hunk 6]
```diff
@@ -1 +1 @@
-from .BlogSerializer import BlogSerializer
\ No newline at end of file
+from app.cms.serializers.BlogSerializer import BlogSerializer
```
Comment (MohamedMegdich1) [inline app/cms/serializers/__init__.py:1]: remove this

File: app/cms/serializers/SliderImageSerializer.py [hunk 7]
```diff
@@ -1,7 +1,9 @@
 from rest_framework import serializers
+
 from app.cms.models import SliderImage
 
+
 class SliderImageSerializer(serializers.ModelSerializer):
```
Comment (MohamedMegdich1) [inline app/cms/serializers/SliderImageSerializer.py:6]: add docstring

File: app/cms/serializers/BlockSerializer.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
+from rest_framework import serializers
+
+from app.cms.models import Block
+
+from .SliderImageSerializer import SliderImageSerializer
+
+
+class BlockSerializer(serializers.ModelSerializer):
```
Comment (MohamedMegdich1) [inline app/cms/serializers/BlockSerializer.py:8]: add docstring

File: app/cms/controllers/__init__.py [hunk 9]
```diff
@@ -1 +1 @@
-from .BlogViewSet import BlogViewSet
\ No newline at end of file
+from .BlogViewSet import BlogViewSet
```
Comment (MohamedMegdich1) [inline app/cms/controllers/__init__.py:1]: remove this

File: app/cms/admin/BlogAdmin.py [hunk 10]
```diff
@@ -1,10 +1,13 @@
-from app.cms.models import Blog
 from django.contrib import admin
-from app.cms.admin.utils.BlockAdmin import BlockInline
+
+from app.cms.admin.BlockAdmin import BlockInline
+from app.cms.models import Blog
+
 
 class BlogAdmin(admin.ModelAdmin):
```
Comment (MohamedMegdich1) [inline app/cms/admin/BlogAdmin.py:7]: docstring

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

**Review by MohamedMegdich1 — APPROVED** [review]

**Pull request merged**